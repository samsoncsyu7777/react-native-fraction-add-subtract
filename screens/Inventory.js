import React, { useState, useContext, useEffect } from "react";
import {
  StyleSheet,
  Text,
  View,
  Platform
} from "react-native";
import theme from "../Theme";
import Icon from 'react-native-vector-icons/EvilIcons';
import { withTheme, TextInput } from 'react-native-paper';
import Item from "./Item";
import secret from "../Secret";
import { UserContext } from "../contexts/UserContext";

function Inventory(props) {
  const { user, setUser } = useContext(UserContext);
  const [upc , setUpc] = useState('');

  const submit = (e) => {
    e.preventDefault();
    searchUpc();
  };

  const searchUpc = () => {
    const PROXY_URL = (Platform.OS === 'web') ? 'https://cors-anywhere.herokuapp.com/' : '';
    const URL = secret.URL+'/api/v1/item/get-upc/'+upc;  

    const res = fetch(PROXY_URL+URL, {
      method: "get",
      headers: { 
        "Content-Type": "application/json",
        "Authorization": "Bearer " + user.token
      }
    })
      .then((response) => response.json())
      .then((responseJson) => {
        if ("content" in responseJson) {
          props.props.navigation.navigate('Item', {itemDTO: responseJson.content[0]});
        } else if ("message" in responseJson) {
          setMessage(responseJson.message);
          setSeverity("error");
        } else if ("errors" in responseJson) {
          setMessage(
            responseJson.errors[0].param + ": " + responseJson.errors[0].msg
          );
        }
      })
      .catch((error) => {
        console.error(error);
      });
  }

    return (
      <View style={styles.container}>
        <Text style={styles.text}>UPC</Text>
        <TextInput
          mode="outlined"
          value={upc}
          label="UPC"
          style={styles.textInput} 
          onChangeText={upc => setUpc(upc)}
        />

        <Icon 
          name="arrow-right" 
          size={90} 
          color="#1690aa"
          onPress={submit}
        />
      </View>
    );
  
}

export default withTheme(Inventory);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#ffffcc",
    alignItems: "center",
    justifyContent: "center"
  },

  text: {
    fontSize: 28,
    fontWeight: "bold",
    color: theme.colors.primary,
    padding: 10
  },

  textInput: {
    width: 270,
    fontSize: 18,
    fontWeight: "bold",
    backgroundColor: theme.colors.background,
    borderColor: theme.colors.primary,
    borderRadius: 40,
    borderWidth: 2,
    padding: 10,
    marginBottom: 20
  },

});
