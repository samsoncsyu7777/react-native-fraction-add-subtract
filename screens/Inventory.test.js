import React from 'react';
import renderer from 'react-test-renderer';

import Inventory from "./Inventory";

describe('<Inventory />', () => {
  it('has 3 children', async () => {
    const tree = renderer.create(<Inventory />).toJSON();
    expect(tree.children.length).toBe(3);
  });

  it('App renders without crashing', async () => {
    const rendered = renderer.create(<Inventory />).toJSON();
    expect(rendered).toBeTruthy();
  });

  it('App test against snapshot', async () => {
    const tree = renderer.create(<Inventory />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('View properties', async () => {
    const tree = renderer.create(<Inventory />).toJSON();
    expect(tree.type).toBe('View');
    expect(tree.props).toHaveProperty('style');
  });

  test('Inventory', async () => {
    expect(Inventory).toHaveProperty('displayName');
  });
    
});