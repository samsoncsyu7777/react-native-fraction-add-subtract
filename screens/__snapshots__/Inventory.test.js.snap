// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Inventory /> App test against snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffcc",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#2080aa",
        "fontSize": 28,
        "fontWeight": "bold",
        "padding": 10,
      }
    }
  >
    UPC
  </Text>
  <View
    style={
      Object {
        "borderColor": "#2080aa",
        "borderRadius": 40,
        "borderWidth": 2,
        "marginBottom": 20,
        "padding": 10,
        "width": 270,
      }
    }
  >
    <View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 6,
            },
            Object {
              "backgroundColor": "#ccfeff",
              "borderColor": "rgba(0, 0, 0, 0.54)",
              "borderRadius": 4,
              "borderWidth": 1,
            },
          ]
        }
      />
      <View
        style={
          Object {
            "minHeight": 64,
            "paddingBottom": 0,
            "paddingTop": 8,
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "backgroundColor": "#ccfeff",
                "bottom": 0,
                "height": 2,
                "left": 10,
                "opacity": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
                "width": 8,
              }
            }
          />
          <Text
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "#ccfeff",
                "color": "transparent",
                "fontFamily": "System",
                "fontSize": 18,
                "fontWeight": "bold",
                "left": 18,
                "opacity": 0,
                "paddingHorizontal": 0,
                "position": "absolute",
                "textAlign": "left",
                "top": 37,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                  Object {
                    "scaleY": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            UPC
          </Text>
          <Text
            numberOfLines={1}
            onLayout={[Function]}
            style={
              Object {
                "color": "#6200ee",
                "fontFamily": "System",
                "fontSize": 18,
                "fontWeight": "bold",
                "left": 0,
                "opacity": 0,
                "paddingHorizontal": 14,
                "position": "absolute",
                "textAlign": "left",
                "top": 36,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            UPC
          </Text>
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontFamily": "System",
                "fontSize": 18,
                "fontWeight": "bold",
                "left": 0,
                "opacity": 0,
                "paddingHorizontal": 14,
                "position": "absolute",
                "textAlign": "left",
                "top": 36,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            UPC
          </Text>
        </View>
        <TextInput
          allowFontScaling={true}
          editable={true}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder=""
          placeholderTextColor="rgba(0, 0, 0, 0.54)"
          rejectResponderTermination={true}
          selectionColor="#6200ee"
          style={
            Array [
              Object {
                "flexGrow": 1,
                "margin": 0,
                "paddingHorizontal": 14,
                "textAlign": "left",
                "zIndex": 1,
              },
              Object {
                "height": 56,
              },
              Object {
                "paddingBottom": 0,
                "paddingTop": 0,
              },
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlignVertical": "center",
              },
              Array [
                Object {},
              ],
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <Text
    allowFontScaling={false}
    onPress={[Function]}
    style={
      Array [
        Object {
          "color": "#1690aa",
          "fontSize": 90,
        },
        undefined,
        Object {
          "fontFamily": "evilicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ï„ƒ
  </Text>
</View>
`;
